src/store/CategoricalDisplayStore.js

```diff
@@ -215,15 +233,15 @@ export default class CategoricalDisplayStore {
     }
     setFavorite(item) {
         if (this.favoritesMap[item.itemId]) {
-            this.favoritesCategory.data = this.favoritesCategory.data.filter(x => x.itemId !== item.itemId);
+            this.favoritesCategory.data = this.favoritesCategory.data.filter(x => x && x.itemId !== item.itemId && x !== item.itemId);
             delete this.favoritesMap[item.itemId];
         } else {
             this.favoritesCategory.data.push(item);
             this.favoritesMap[item.itemId] = true;
         }
 
         const layout = this.context.stx.layout;
-        layout.favorites[this.favoritesId] = toJS(this.favoritesCategory.data);
+        layout.favorites[this.favoritesId] = toJS(this.favoritesCategory.data).filter(item => item).map( item => typeof item === 'string' ? item : item.itemId);
```

Comment:
@shayanypn why do we need `.filter(item => item)` ?

Reply:
@aminroosta is some cases (ex, user work with old version which store whole object) by adding new fav, and storing it, the code generate some `null` item, that make store crash. just to prevent this crash

Reply:
@shayanypn Ok, that's good